vllm/model_executor/models/mamba_cache.py

```diff
@@ -0,0 +1,243 @@
+from typing import Dict, List, Optional
```

Comment:
I think it would be nice to move MambaCacheManager (Mamba-specific class) to model_executor/layers/mamba, soon enough we will also have a generic Mamba layer there , WDYT?

Reply:
I agree that this file does not belong here, but I don't think we should put it in `model_executor`, since it doesn't actually do any execution.

Ultimately, I think it should live in a file named something like `vllm/core/mamba_block_manager.py`, alongside `block_manager`, and we should put in a refactor so that mamba state natively lives alongside attention. That will be much simpler if we land https://github.com/vllm-project/vllm/pull/7428 first.

I'm somewhat inclined to leave this file here in the meantime.